Description: Memory leak at line in metadata_iterators.c, 'node' is not freed.
From: https://sourceforge.net/tracker/?func=detail&aid=2946736&group_id=13478&atid=313478

(from Debian Wheezy)

---
 src/libFLAC/metadata_iterators.c |    1 +
 1 file changed, 1 insertion(+)

--- a/src/libFLAC/metadata_iterators.c
+++ b/src/libFLAC/metadata_iterators.c
@@ -1217,6 +1217,7 @@ static FLAC__bool chain_read_cb_(FLAC__M
 			}
 
 			if(!read_metadata_block_header_cb_(handle, read_cb, &is_last, &type, &length)) {
+				node_delete_(node);
 				chain->status = FLAC__METADATA_CHAIN_STATUS_READ_ERROR;
 				return false;
 			}
